### PR TITLE
[bitnami/consul] Add Consul Helm Chart tests

### PR DESCRIPTION
**Pull request by superaleks**
**Description of the change**
Increasing the test coverage of the Grafana Loki Helm Chart by adding Goss tests. 

**Benefits**

Ensuring higher quality of the Helm charts.

**Possible drawbacks**
Automated tests could show to be potentially flaky. 

**Test results**

:gear: - [PASS](https://github.com/superaleks/charts/pull/35)

### Checklist

- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in the agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by FraPazGal — COMMENTED** [review]
Thanks for the PR @superaleks! I added a few comments to your changes, could you look at them?

File: .vib/consul/cypress/cypress/integration/consul_spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    cy.fixture('key').then((key) => {
+      cy.contains(`${key.newKey.testKey}.${random}`);
+    });
+  });
+});
+
+it('shows configured data center', () => {
+  cy.visit('/');
+  cy.get('[role="banner"]').contains(Cypress.env('datacenterName'));
+  cy.contains(Cypress.env('datacenterName'));
```
Comment (joancafom) [inline .vib/consul/cypress/cypress/integration/consul_spec.js:47]: Isn't this the same thing? Line 47 will also potentially match line 46

File: .vib/consul/goss/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - '::'
+file:   
+  /opt/bitnami/consul/conf/consul.json:
+    mode: "0644"
+    filetype: file
+    contains: 
+      - "8500"
+      - "8600"
+      - "8400"
+      - "8301"
```
Comment (joancafom) [inline .vib/consul/goss/goss.yaml:14]: You may use environment variables (if defined) in favor of fixed values.

Also, these seem like ports for different protocols. If the configuration file defines them as something like `http: 8500` it may be interesting to check the protocols are assigned the right port:

```
{{ printf "/http.*%d/" .Env.CONSUL_HTTP_PORT_NUMBER }}
```

File: .vib/consul/goss/goss.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+port:
+  tcp6:8400:
```
Comment (joancafom) [inline .vib/consul/goss/goss.yaml:2]: Instead of a fixed `8400` value for the port, you may use the specific env. var. to adapt to changes

File: .vib/consul/goss/goss.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    exit-status: 0
+    stdout: []
+    stderr: []
+  check-env:
+    exec: env
+    exit-status: 0
+    stdout: 
+      - "HOSTNAME=consul-0"
+      - "CONSUL_DATACENTER=datacenter1"
+    stderr: []
```
Comment (joancafom) [inline .vib/consul/goss/goss.yaml:28]: I would like to hear your opinion on this, but to me, usually there is no such a big return in checking environment variables are set: they are just a way to pass down information to our configuration scripts.

Instead, if we want to assert that `datacenter1` is effectively the name of the datacenter I would resort to other means when possible.

TL;DR
An environment variable having a specific value does not guarantee the logic has correctly applied it.

File: .vib/consul/cypress/cypress/integration/consul_spec.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/// <reference types="cypress" />
+
+import { random } from './utils';
+
+it('shows all instances of services', () => {
+  cy.visit('/ui/datacenter1/services');
+  cy.contains('Services');
+  cy.get('.linkable').contains('consul');
+  cy.get('.detail span').contains('3 instances');
```
Comment (joancafom) [inline .vib/consul/cypress/cypress/integration/consul_spec.js:9]: I believe that having 3 instances is entirely dependent on installation parameters (`replicaCount`). I would add it to the `runtime_params` & `cypress.json` env var section, as we are actively checking this value

File: .vib/consul/cypress/cypress/integration/consul_spec.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+it('is able to display all of the nodes', () => {
+  cy.visit('/ui/datacenter1/nodes');
+  cy.get('.ember-view').within(() => {
+    cy.get('.passing').should('have.length', 3);
+    cy.get('.passing').first().click();
+  });
+  cy.contains('Agent alive and reachable');
+  cy.contains('Service Instances').click();
+  cy.get('.ember-view').within(() => {
+    cy.get('.address').contains('8400');
```
Comment (joancafom) [inline .vib/consul/cypress/cypress/integration/consul_spec.js:59]: This is also dependent on installation params, `containerPorts.rpc`. I would proceed the same, adding it to the `runtime_params` and `cypress.json` file

**Review by joancafom — COMMENTED** [review]
Thanks for the PR @superaleks ! I left you some thoughts 😁

File: .vib/consul/cypress/cypress/integration/consul_spec.js (hunk 1 above)
Comment (superaleks) [reply .vib/consul/cypress/cypress/integration/consul_spec.js]: True!

File: .vib/consul/goss/goss.yaml (hunk 4 above)
Comment (superaleks) [reply .vib/consul/goss/goss.yaml]: I see your point, I've replaced this test with one that seems to be more useful.

File: .vib/consul/cypress/cypress.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "baseUrl": "http://localhost",
+  "pageLoadTimeout": 240000,
+  "defaultCommandTimeout": 60000,
+  "env": {
+    "datacenterName": "datacenter1",
+    "replicaCount": "3",
+    "containerPorts": "8400"
```
Comment (joancafom) [inline .vib/consul/cypress/cypress.json:8]: This is a specific port (RPC), hence I would rename the env to something like `containerRpcPort`

Comment (superaleks) [reply .vib/consul/cypress/cypress.json]: Changed!

**Review by joancafom — APPROVED** [review]
LGTM!

**Review by FraPazGal — APPROVED** [review]
